lalrpop/src/build/mod.rs

```diff
@@ -205,6 +205,35 @@ fn needs_rebuild(lalrpop_file: &Path, rs_file: &Path) -> io::Result<bool> {
     }
 }
 
+/// Handles a [walkdir::Error] if the root cause is a dangling symlink.
+///
+/// Returns `Ok` if the error could be handled, otherwise returns `Err(err)`.
+fn handle_dangling_symlink_error(err: walkdir::Error) -> Result<(), walkdir::Error> {
+    let is_not_found = err.io_error().map(|io_err| io_err.kind()) == Some(io::ErrorKind::NotFound);
```

Comment:
Do we also want to gracefully handle symlink cycles here with [loop_ancestor](https://docs.rs/walkdir/latest/walkdir/struct.Error.html#method.loop_ancestor)?

Reply:
I believe no; I haven't tried but I think we'll print the error message according generated by `walkdir` and exit. Tbh I'm not sure I want to think too much about something we could do in case of cycles; wouldn't we need to recreate the cyclic symlinks also in the target directory so that every `.lalrop` file in the source directory is accessible under the same relative paths also in the target directory? I suggest we refuse compiling such crates for now, and if somebody has a real need for this we can think about it again.

Reply:
Or do you mean by "gracefully" handling it that we should print a reasonable error, and not that we should try to proceed with lalrpop code generation?

Reply:
Hmm, fair enough. It doesn't look like normal `io::Error`'s can be converted into `walkdir::Error` so it's a bit more work to propagate this information up the stack.

We could raise an warning and try to keep compiling, but your right that that could get wonky.